tensorflow_addons/metrics/r_square.py

```diff
@@ -70,23 +107,39 @@ def update_state(self, y_true, y_pred, sample_weight=None):
         sample_weight = weights_broadcast_ops.broadcast_weights(sample_weight, y_true)
 
         weighted_y_true = tf.multiply(y_true, sample_weight)
-        self.sum.assign_add(tf.reduce_sum(weighted_y_true))
-        self.squared_sum.assign_add(tf.reduce_sum(tf.multiply(y_true, weighted_y_true)))
+        self.sum.assign_add(tf.reduce_sum(weighted_y_true, axis=0))
+        self.squared_sum.assign_add(
+            tf.reduce_sum(tf.multiply(y_true, weighted_y_true), axis=0)
+        )
         self.res.assign_add(
             tf.reduce_sum(
-                tf.multiply(tf.square(tf.subtract(y_true, y_pred)), sample_weight)
+                tf.multiply(tf.square(tf.subtract(y_true, y_pred)), sample_weight),
+                axis=0,
             )
         )
-        self.count.assign_add(tf.reduce_sum(sample_weight))
+        self.count.assign_add(tf.reduce_sum(sample_weight, axis=0))
 
-    def result(self):
+    def result(self) -> tf.Tensor:
         mean = self.sum / self.count
         total = self.squared_sum - self.sum * mean
-        return 1 - (self.res / total)
+        raw_scores = 1 - (self.res / total)
+
+        if self.multioutput == "raw_values":
+            return raw_scores
+        elif self.multioutput == "uniform_average":
+            return tf.reduce_mean(raw_scores)
+        elif self.multioutput == "variance_weighted":
+            return _reduce_average(raw_scores, weights=total)
+        else:
+            raise RuntimeError(
+                "The multioutput attribute must be one of {}, but was: {}".format(
+                    VALID_MULTIOUTPUT, self.multioutput
+                )
+            )
 
-    def reset_states(self):
+    def reset_states(self) -> None:
         # The state of the metric will be reset at the start of each epoch.
-        self.squared_sum.assign(0.0)
-        self.sum.assign(0.0)
-        self.res.assign(0.0)
-        self.count.assign(0.0)
+        self.squared_sum.assign(tf.zeros(shape=self.y_shape, dtype=self.dtype))
```

Comment:
Is this necessary? I would have expected tensorflow to gracefully broadcast the `0.0`. I'm disapointed if TF doesn't do it automatically.

Reply:
Just me being conservative. Reverted.